pants

```diff
@@ -65,8 +98,14 @@ export PY="${PY:-python3}"
 # Pants will try to use Python 2 for subprocesses. Without setting minor version constraints, when using Python 3
 # we could end up using a different Python minor version for subprocesses than we do for Pants.
 py_major_minor_patch=$(${PY} -c 'import sys; print(".".join(map(str, sys.version_info[0:3])))')
+
 export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS="${PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS:-['CPython==${py_major_minor_patch}']}"
 
+# We can currently assume the CI environment does not switch python versions within a shard.
+if [[ "${ONLY_USING_SINGLE_PYTHON_VERSION:-false}" != 'true' ]]; then
```

Comment:
Danny and I were talking over Slack about alternatively naming this `DISABLE_AUTO_CLEAN_ALL`. We have to decide if we want to describe the property the environment has or what we're specifically doing to it.

Preferences from others?

Reply:
I think if we used `DISABLE_AUTO_CLEAN_ALL` and then want to do something completely different (such as eventually not `clean-all`), we would have to change the variable name, and it runs the danger of making this confusing part of our init more confusing. I think describing the reason why we're not doing the clean-all, then leaving a comment (which itself can be updated) about what that means is a more stable approach.